### PR TITLE
LDAP Password Modify Extended Operation support

### LINKED ISSUE #14135: LDAP Password Modify Extended Operation
Password change of LDAP users should be done using Password Modify Extended Operation (RFC 3062), which is available in PHP since version 7.2.

### PR DESCRIPTION
**Pull request by army1349**
Fixes #14135

### CODE REVIEW

**Comment by blizzz** [comment]
Like this it could go in only, if we decide to drop PHP 7.1 support in Nextcloud 17.  cc @rullzer  That said, there is still some time before 16 is being branched off and we merge features again, let's see then again whether this will be ok or could use a function_exists check instead.

**Comment by army1349** [comment]
@blizzz function_exists check is already done. This should not fail on older PHP.

File: apps/user_ldap/lib/LDAP.php [hunk 1]
```diff
@@ -219,6 +219,17 @@ public function modReplace($link, $userDN, $password) {
...
 	}
 
+	/**
+	 * @param LDAP $link
+	 * @param string $userDN
+	 * @param string $oldPassword
+	 * @param string $password
+	 * @return bool
+	 */
+	public function exopPasswd($link, $userDN, $oldPassword, $password) {
```
Comment (kesselb) [inline apps/user_ldap/lib/LDAP.php]:
```suggestion
	public function exopPasswd($link, $userDN, $oldPassword, $password): bool {
```

Comment (blizzz) [reply apps/user_ldap/lib/LDAP.php]: Essentially yes, but then again the whole file looks like that, so we can keep it as is (and polish it in seperate PR)

Comment (blizzz) [reply apps/user_ldap/lib/LDAP.php]: This should be added to ILDAPWrapper, but it is not a big deal, really.

**Review by blizzz — APPROVED** [review]
Works, and even better since the value ends up hashed out of the box.

**Review by MorrisJobke — APPROVED** [review]
Code makes sense 👍

**Pull request merged**